src/compiler/types.ts

```diff
@@ -885,6 +885,144 @@ namespace ts {
         /* @internal */ jsDocCache?: readonly JSDocTag[];     // Cache for getJSDocTags
     }
 
+    /* @internal */
+    export type HasChildren =

```

Comment:
What happens if someone adds a new syntax kind and forgets to update this?  Seems like their new node might not emit its children?

Reply:
If both this and #50225 use `HasChildren`, then you wouldn't be able to define a `forEachChild` branch or `visitEachChild` branch without adding an entry, and if you add an entry to `HasChildren` you would have compile time errors if you forget to add branches to `forEachChild` and `visitEachChild`.